src/dbt_osmosis/core/osmosis.py

```diff
@@ -336,15 +341,15 @@ def get_columns(self, parts: Tuple[str, str, str]) -> List[str]:
                 if model.split(".")[-1] == parts[-1]
             ]
             if matching_models:
-                return [col.lower() for col in matching_models[0]["columns"].keys()]
+                return {self.column_casing(col['name']): ColumnMetadata(**col) for col in matching_models[0]["columns"].values()}
```

Comment:
Hi! Do we need self.column_casing for the column name inside the returned ColumnMetadata?

Reply:
added